config/application.rb

```diff
@@ -52,19 +52,14 @@
 require_relative '../lib/stoplight/redis_data_store_extensions'
 require_relative '../lib/active_record/database_tasks_extensions'
 require_relative '../lib/active_record/batches'
-require_relative '../lib/active_record/with_recursive'
-require_relative '../lib/arel/union_parenthesizing'
 require_relative '../lib/simple_navigation/item_extensions'
 
 Bundler.require(:pam_authentication) if ENV['PAM_ENABLED'] == 'true'
 
 module Mastodon
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 7.1
-
-    # Explicitly set the cache format version to align with Rails version
-    config.active_support.cache_format_version = 7.1
+    config.load_defaults 7.2
```

Comment:
Going directly to this instead of the new_framework_defaults file, because its very minimal and has nothing for us to step through really (or rather, everything is either desirable or not relevant) - https://github.com/rails/rails/blob/v7.2.1.1/railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_2.rb.tt